packages/composite-checkout/src/components/billing-fields.js

```diff
@@ -0,0 +1,552 @@
+/**
+ * External dependencies
+ */
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+
+/**
+ * Internal dependencies
+ */
+import { useLocalize } from '../lib/localize';
+import { useCheckoutLineItems } from '../index';
+import GridRow from './grid-row';
+import Field from './field';
+
+export default function BillingFields( {
+	summary,
+	setPaymentData,
+	paymentData,
+	isActive,
+	isComplete,
+} ) {
+	const [ items ] = useCheckoutLineItems();
+	const [ showDomainContactFields, setShowDomainContactFields ] = useState( false );
+
+	if ( summary && isComplete ) {
+		return <BillingFormSummary paymentData={ paymentData } />;
```

Comment:
If it's actually not being changed, you could try the function form of setState:

```
const toggleDomainFieldsVisibility = () => setShowDomainContactFields(current => ! current);
```

Reply:
the toggle works for the UI in the `return` part of this method. The value doesn't seem to change if I pass it as a prop to `BillingFormSummary`. Does that make sense?

Reply:
Oh! I have a feeling I know what you're talking about. Remember that there's _two_ instances of `BillingFields` on the page; the expanded one and the summary one. Any call to `useState()` inside this component will only apply that _that copy_ of the component. So if you toggle the state in the expanded form and then switch it so that the summary is visible, the state in the visible one will not be toggled.

If you want to have state that is shared between components, we'll need to move it up the component tree to the nearest common parent, or use a very high state like the data stored in `paymentData`. 🤔

Reply:
Moving it up a level did the trick! I have it working like I wanted now — Thank you!